src/main/java/emissary/core/SafeUsageChecker.java

```diff
@@ -0,0 +1,110 @@
+package emissary.core;
+
+import emissary.config.ConfigUtil;
+import emissary.config.Configurator;
+import emissary.core.channels.SeekableByteChannelFactory;
+import emissary.util.ByteUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility for validating that Places safely interact with IBDO payloads in byte array form. Specifically, this class
+ * helps with validating that changes to a IBDO's payload are followed by a call to the
+ * {@link IBaseDataObject#setData(byte[]) setData(byte[])}, {@link IBaseDataObject#setData(byte[], int, int)
+ * setData(byte[], int, int)}, or {@link IBaseDataObject#setChannelFactory(SeekableByteChannelFactory)
+ * setChannelFactory(SeekableByteChannelFactory)} method.
+ */
+public class SafeUsageChecker {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(SafeUsageChecker.class);
+
+    public static final String ENABLED_KEY = "ENABLED";
+    public static final boolean ENABLED_FROM_CONFIGURATION;
+    public static final String UNSAFE_MODIFICATION_DETECTED = "Detected unsafe changes to IBDO byte array contents";
+
+    static {
```

Comment:
Suggest adding an explanatory comment to explain the choice of initialization pattern, to discourage any future change that removes the static field and collapses the initialization into the default constructor. 

```suggestion
    // to minimize I/O, we only want to read the config file once regardless of the number of instances created
    static {
```